### PR TITLE
Forbid leader election on older Kubernetes clusters

### PR DESCRIPTION
**Pull request by dhaiducek**
Older Kubernetes (i.e. Openshift 3.11) doesn't support the lease API, so this detects the Kubernetes version from the ClusterClaim and forces the replica to 1 on older Kubernetes, even if the user attempts to override it.

ref: https://issues.redhat.com/browse/ACM-5042

This also bumps the Go version and packages, fixes the resulting linting issues, and removes the replacement from a forked repo now that the issue is resolved.

### CODE REVIEW

File: pkg/addon/common.go [hunk 1]
```diff
@@ -141,9 +145,16 @@ func GetAndAddAgent(
 		return fmt.Errorf("failed getting the %v agent addon: %w", addonName, err)
 	}
 
-	agentAddon = &PolicyAgentAddon{agentAddon}
+	clusterClient, err := clusterv1client.NewForConfig(controllerContext.KubeConfig)
+	if err != nil {
+		return err
+	}
+
+	clusterInformers := clusterv1informers.NewSharedInformerFactory(clusterClient, 10*time.Minute)
```
Comment (zhujian7) [inline pkg/addon/common.go:153]: not sure if it is because we did not call clusterInformers.Start(ctx.Done()) , please check.

Comment (dhaiducek) [reply pkg/addon/common.go:154]: This seems to have done it--thanks for the pointer!

```
go clusterInformers.Cluster().V1().ManagedClusters().Informer().Run(ctx.Done())
```

**Review by gparvin — COMMENTED** [review]
What's the difference between the two `mandateValues` functions?  They look really similar.

**Comment by dhaiducek** [comment]
> What's the difference between the two `mandateValues` functions? They look really similar.

They're identical, but one is for `config-policy-controller` and the other is for `governance-policy-framework`.

**Review by JustinKuli — DISMISSED** [review]
LGTM! I really appreciate some of the test updates as well.

`mandateValues` could be a common function, especially if other agents (eg in stolostron) are likely to use it. But the second we need a specific value for a specific agent, we'd be breaking the function up again... so I think it's totally up to you. Only holding in case Gus has a stronger opinion.

/hold

**Comment by gparvin** [comment]
Sounds good.  Thanks!
/unhold

**Comment by dhaiducek** [comment]
The Kubernetes version had a `v`, which isn't part of semantic versioning. However, there's a `ParseTolerant()` function that strips it and handles it properly. Doing some final tests. The one limitation I realized is that this implementation also means that the replicas can't be scaled to zero. I think that's acceptable since the workaround there would be to delete the ManagedClusterAddon and then reapply it.

**Review by gparvin — APPROVED** [review]
Thanks for testing with OCP 3.11!

**Pull request merged**